gtda/diagrams/features.py

```diff
@@ -525,3 +525,133 @@ def transform(self, X, y=None):
+
+    See also
+    --------
+    PersistenceLandscape, PersistenceEntropy, HeatKernel, Amplitude, \
+    PairwiseDistance, BettiCurve, gtda.homology.VietorisRipsPersistence
+
+    Notes
+    -----
+    The samplings in :attr:`samplings_` are in general different between
+    different homology dimensions. This means that the j-th entry of
+    a Silhouette in homology dimension q typically arises from
+    a different parameter values to the j-th entry of a curve
+    in dimension q-1'.
+
+    """
+
+    _hyperparameters = {'order': [float, (1, np.inf)],
+                        'n_values': [int, (1, np.inf)]}
+
+    def __init__(self, order=1, n_values=100, n_jobs=None):
+        self.order = order
+        self.n_values = n_values
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y=None):
+        """Store all observed homology dimensions in
+        :attr:`homology_dimensions_` and, for each dimension separately,
+        store evenly sample filtration parameter values in :attr:`samplings_`.
+        Then, return the estimator.
+        This method is here to implement the usual scikit-learn API and hence
+        work in pipelines.
+        Parameters
+        ----------
+        X : ndarray of shape (n_samples, n_features, 3)
+            Input data. Array of persistence diagrams, each a collection of
+            triples [b, d, q] representing persistent topological features
+            through their birth (b), death (d) and homology dimension (q).
+        y : None
+            There is no need for a target in a transformer, yet the pipeline
+            API requires this parameter.
+        Returns
+        -------
+        self : object
+        """
+        X = check_diagram(X)
+        validate_params(self.get_params(), self._hyperparameters)
+
+        self.homology_dimensions_ = sorted(list(set(X[0, :, 2])))
+        self._n_dimensions = len(self.homology_dimensions_)
+
+        self._samplings, _ = _discretize(X, n_values=self.n_values)
+        self.samplings_ = {dim: s.flatten()
+                           for dim, s in self._samplings.items()}
+
+        return self
+
+    def transform(self, X, y=None):
+        """Compute Silhouettes of diagrams in X.
+
+        Parameters
+        ----------
+        X : ndarray of shape (n_samples, n_features, 3)
+            Input data. Array of persistence diagrams, each a collection of
+            triples [b, d, q] representing persistent topological features
+            through their birth (b), death (d) and homology dimension (q).
+
+        y : None
+            There is no need for a target in a transformer, yet the pipeline
+            API requires this parameter.
+
+        Returns
+        -------
+        Xt : ndarray of shape (n_samples, n_homology_dimensions, n_values)
+            Silhouette: one curve (represented as a one-dimensional array
+            of integer values) per sample and per homology dimension seen
```

Comment:
Not sure I understand why it is an array "of integer values"

Reply:
My bad